src/bash.rs

```diff
@@ -154,19 +149,13 @@ pub fn write_completion_script<W: Write>(buffer: &mut W, command: &str, dfa: &DF
         write!(buffer, r#"
     if [[ -v "specialized_commands[$state]" ]]; then
         local command_id=${{specialized_commands[$state]}}
-        IFS=$'\n' read -r -d '' -a command_completions < <( _{command}_spec_${{command_id}} "${{COMP_WORDS[$COMP_CWORD]}}" && printf '\0' )
-        for line in "${{command_completions[@]}}"; do
-            local elem=$(echo "$line" | cut -f1)
-            completions+=($elem)
-        done
+        completions+="$(_{command}_spec_${{command_id}} "${{COMP_WORDS[$COMP_CWORD]}}" | cut -f1)"$'\n'
     fi
-
 "#)?;
     }
 
     write!(buffer, r#"
-    completions=${{completions[@]}}
-    COMPREPLY=($(compgen -W "$completions" -- "${{COMP_WORDS[$COMP_CWORD]}}"))
+    mapfile -t COMPREPLY < <(IFS=$'\n' compgen -W "$completions" -- "${{COMP_WORDS[$COMP_CWORD]}}")
```

Comment:
`mapfile` (aka `readarray`) and `IFS=$'\n'` are needed to prevent word splitting on spaces.

Reply:
What I meant here was really `completions=${{completions[*]}}`.  Having an array with completions instead of a single string is more flexible.

Reply:
The `mapfile -t COMPREPLY …` line should be fine though, provided the e2e tests pass